### DISCUSSION TITLE
How do we notify the changes in dir to windows in dokan or dokanfuse?

### DISCUSSION THREAD

**Discussion by zqlustc** [dokan-dev/dokany#977] [Q&A] [answered]
In the samba filesystem:
The client opened a dir namend "test", If the server changed the dir(create or delete a file), the client dir wll be updated immediately(it is not necessary to right click and refresh or cpress F5)

In the dokan mount point:
If the file is deleted by others in the client, we have to  right click and refresh or cpress F5 to refresh current dir. 

How to solve the problem in the dokan or something could solve the problem in the dokan but I do not know ? @Liryna

**Comment by Liryna** [comment, marked as answer]
You can do this by using the notification functions that you can find in the dokan.h file.

They are not directly available from dokanfuse so it will need some code changes. If you are going to make the changes, please do not hesitate to create a pull request.

**Reply by zqlustc** [reply]
I have tried the scene:
 When I delete a file(named test.txt) failed in fuseunlink(now the file view have been deleted in the windows file explore), I call the DokanNotifyUpdate/DokanNotifyCreate through dokanfuse to notify to update the current dir. But the current windows file explore will not updates the file view(test.txt) immediately and I have to right click refresh to updates. logs showed the notify was success, but the file view won't show up immediately.

**Reply by Liryna** [reply]
Could you provide an example of call & params for `DokanNotifyUpdate/DokanNotifyCreate` ?
